recipes/onnxruntime/all/conanfile.py

```diff
@@ -0,0 +1,233 @@
+from conan.tools.apple import is_apple_os
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.env import VirtualBuildEnv
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class OnnxRuntimeConan(ConanFile):
+    name = "onnxruntime"
+    description = "ONNX Runtime: cross-platform, high performance ML inferencing and training accelerator"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    homepage = "https://onnxruntime.ai"
+    topics = ("deep-learning", "onnx", "neural-networks", "machine-learning", "ai-framework", "hardware-acceleration")
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("abseil/20230125.2")
+        self.requires("protobuf/3.21.9")
+        self.requires("date/3.0.1")
+        self.requires("re2/20230301")
+        self.requires("onnx/1.13.1")
+        self.requires("flatbuffers/1.12.0")
+        self.requires("boost/1.81.0")  # for mp11
+        self.requires("safeint/3.0.28")
+        self.requires("nlohmann_json/3.11.2")
+        self.requires("eigen/3.4.0")
+        self.requires("ms-gsl/4.0.0")
+        self.requires("cpuinfo/cci.20220228")
+        if self.settings.os != "Windows":
+            self.requires("nsync/1.25.0")
+        else:
+            self.requires("wil/1.0.230202.1")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def _cmake_new_enough(self, required_version):
```

Comment:
This is not required, we prefer the graph to be less dependent on the users setup

Reply:
This might not be required but the rational for it existing in https://github.com/conan-io/conan-center-index/pull/15231 just does make sense.

Reply:
Please See #17192 along with existing conan features this is not the best practice :)

Reply:
I dropped _cmake_new_enough since it was done in a few recipes but I still think the rational for it existing in https://github.com/conan-io/conan-center-index/pull/15231 just does make sense. It might make sense to do this differently than duplicating code across all recipes though (maybe consider the `cmake` recipe as a special citizen in conan itself and do the check there ? maybe have another recipe `virtual-cmake` that does the check ?)

Reply:
Thats completely not required.

 you can have a version range and use the `system_tool` to tell conan what you are providing

Cmake is not special, the client can solve this in a generic manner

Reply:
Thanks I was missing the pointer to find that in the docs.